TSPL.docc/LanguageGuide/Concurrency.md

```diff
@@ -564,33 +637,192 @@ the explicit parent-child relationships between tasks
 let Swift handle some behaviors like propagating cancellation for you,
 and lets Swift detect some errors at compile time.
 
+[`TaskGroup`]: https://developer.apple.com/documentation/swift/taskgroup
+
+<!-- TR: What's a good example of this kind of error? -->
+
+Here's another version of the code to download photos
+that handles any number of photos:
+
 ```swift
 await withTaskGroup(of: Data.self) { taskGroup in
+    let photoNames = await listPhotos(inGallery: "Summer Vacation")
+    for name in photoNames {
+        taskGroup.addTask {
+            let photo = downloadPhoto(named: name)
+            show(photo)
+        }
+    }
+}
+```
+
+The code above creates a new task group,
+and then creates child tasks of that task group
+to download each photo is the gallery.
+Each task group downloads a single photograph
+and then adds it to the photographs being displayed.
+Because there isn't a specific priority passed
+when calling `addTask(priority:operation:)`,
+these child tasks tasks get the same priority as the task group.
+Swift schedules these tasks,
+running as many of them at the same time as conditions allow.
+
+> Note:
+> If the code to download a photo could throw an error,
+> you would call `withThrowingTaskGroup(of:returning:body:)` instead.
+
+In the code listing above,
+each photo is downloaded and immediately displayed,
+so the task group doesn't return any results.
+For a task group that returns a result,
+you add code that accumulates its result
+inside the closure you pass to `withTaskGroup(of:returning:body:)`.
+
+```
+let photos = await withTaskGroup(of: Data.self) { taskGroup in
     let photoNames = await listPhotos(inGallery: "Summer Vacation")
     for name in photoNames {
         taskGroup.addTask { await downloadPhoto(named: name) }
     }
+
+    var results: [Data] = []
+    for await photo in taskGroup {
+        results.append(photo)
+    }
+    return results
```

Comment:
This is good, but i'd show it earlier -- this is the "normal" way to use a group.

You could instead first show:

```
     for await photo in group {
         show(photo)
     }
```

and then the 

```
    var results: [Data] = []
    results.reserveCapacity(photoNames.size)
     for await photo in group {
         results.append(photo)
     }
     return results
```

maybe?